samples/show-scale-bar/src/main/java/com/esri/arcgismaps/sample/showscalebar/screens/ShowScaleBarScreen.kt

```diff
@@ -0,0 +1,245 @@
+                        onScalebarStyleSelected = { currentScalebarStyle = it },
+                        onDismissRequest = { isScalebarDialogOptionsVisible = false },
+                        onAutoHideDelaySelected = { currentAutoHideDelay = it },
+                        onGeodeticCalculationsToggled = {
+                            isGeodeticCalculationsEnabled = !isGeodeticCalculationsEnabled
+                        }
+                    )
+                }
+            }
+            mapViewModel.messageDialogVM.apply {
+                if (dialogStatus) {
+                    MessageDialog(
+                        title = messageTitle,
+                        description = messageDescription,
+                        onDismissRequest = ::dismissDialog
+                    )
+                }
+            }
+        }
+    )
+}
+
+@Composable
+fun ScalebarDialogOptions(
+    currentScalebarStyle: ScalebarStyle,
+    currentScalebarUnitSystem: UnitSystem,
+    currentAutoHideDelay: Duration,
+    onDismissRequest: () -> Unit,
+    isGeodeticCalculationsEnabled: Boolean,
+    onScalebarStyleSelected: (ScalebarStyle) -> Unit,
+    onScalebarUnitSystemSelected: (UnitSystem) -> Unit,
+    onAutoHideDelaySelected: (Duration) -> Unit,
+    onGeodeticCalculationsToggled: (Boolean) -> Unit
+) {
+    // List of all the supported scale bar properties
+    val scalebarStyles = ScalebarStyle.entries.toList()
+    val scalebarUnitSystems = listOf(UnitSystem.Metric, UnitSystem.Imperial)
+    val autoHideDelays = listOf(
+        Duration.INFINITE,
+        1.toDuration(DurationUnit.SECONDS),
+        3.toDuration(DurationUnit.SECONDS),
+        5.toDuration(DurationUnit.SECONDS)
+    )
+
+    BasicAlertDialog(onDismissRequest = onDismissRequest) {
+        Column(
+            modifier = Modifier
+                .clip(RoundedCornerShape(12.dp))
+                .background(MaterialTheme.colorScheme.background)
+                .padding(12.dp)
+                .fillMaxWidth()
+                .verticalScroll(rememberScrollState()),
+            verticalArrangement = Arrangement.spacedBy(12.dp),
+            horizontalAlignment = Alignment.CenterHorizontally,
+        ) {
+            Text("Scale bar options: ", style = MaterialTheme.typography.titleMedium)
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.SpaceBetween,
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Text(
+                    text = "Enable Geodetic Calculations",
+                    style = MaterialTheme.typography.labelLarge
+                )
+                Switch(
+                    checked = isGeodeticCalculationsEnabled,
+                    onCheckedChange = onGeodeticCalculationsToggled
+                )
+            }
+            DropDownMenuBox(
+                textFieldLabel = "Scale bar style: ",
```

Comment:
```suggestion
                textFieldLabel = "Scale bar style:",
```

These and following texts have trailing whitespace